main.py

```diff
@@ -131,73 +145,147 @@ def calculate_interest(self, rate: float) -> None:
         print("Debit accounts do not earn interest.")
 
 
-class Customer:
-    def __init__(self, account_number: str, account: Account) -> None:
+class SavingsAccount:
```

Comment:
This class should extend the Account class like the other specific account type classes, a lot of the code is unchanged and repeated from Account